objectstore-service/src/metadata.rs

```diff
@@ -0,0 +1,46 @@
+use serde::Deserialize;
+use std::time::Duration;
+
+/// Denotes the storage usecase
+///
+/// Different decisions can be made based on this usecase, such as using a different underlying
+/// storage backend.
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum Usecase {
+    Attachments,
```

Comment:
Let us make `UseCase` wrap a plain String - if we care about `Copy` we can swap this out with an interned string. 

We need to allow produce use cases to be added without code changes of our service or ideally the client library. Ideally, use cases are pure configuration supplied to our service at startup. If we want to type use cases in the client, I would suggest we expose this from sentry_protos or derive a type from the place where the use cases are configured.

Note that this doesn't have to be pure runtime-config, our service should just be agnostic over _specific_ use cases.